core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -106,3 +109,196 @@ func testHeaderRLPBackwardsCompatibility(t *testing.T) {
 		assert.Equal(t, hdr, got)
 	})
 }
+
+func TestBodyRLPBackwardsCompatibility(t *testing.T) {
+	rng := ethtest.NewPseudoRand(0)
+
+	randTx := func() *Transaction { return NewTx(&LegacyTx{Nonce: rng.Uint64()}) }
+	randHdr := func() *Header { return &Header{ParentHash: rng.Hash()} }
+	randWithdraw := func() *Withdrawal { return &Withdrawal{Index: rng.Uint64()} }
+
+	// We build up test-case [Body] instances from the power set of each of
+	// these components.
+	txs := [][]*Transaction{
+		nil, {}, // Must be equivalent for non-optional field
+		{randTx()},
+		{randTx(), randTx()}, // Demonstrates nested lists
+	}
+	uncles := [][]*Header{
+		nil, {},
+		{randHdr()},
+		{randHdr(), randHdr()},
+	}
+	withdrawals := [][]*Withdrawal{
+		nil, {}, // Must be different for optional field
+		{randWithdraw()},
+		{randWithdraw(), randWithdraw()},
+	}
+
+	var bodies []*Body
+	for _, tx := range txs {
+		for _, u := range uncles {
+			for _, w := range withdrawals {
```

Comment:
nit slightly confusing var naming, maybe rename:

- `txs` -> `txsMatrix`
- `tx` -> `txs`
- `uncles` -> `unclesMatrix`
- `u` -> `uncles`
- `withdrawals` -> `withdrawalsMatrix`
- `w` -> `withdrawals`

Reply:
Done on the matrices but not the loop variables as they're only used on one line.